### PR TITLE
Fix #494

### LINKED ISSUE #494: Parse repeated by-name parameters in Dotty dialect
Repeated by-name parameters:

``` scala
class Foo(bars: => Int*)
```

Reference: http://docs.scala-lang.org/sips/pending/repeated-byname.html

Scala meta has the concept of dialect for different Scala syntax (think sbt, scala or dotty). The dotty dialect does not yet support repeated by-name parameters.

A good place to start is https://github.com/scalameta/scalameta/blob/master/scalameta/parsers/src/main/scala/scala/meta/internal/parsers/ScalametaParser.scala#L2680.

### PR DESCRIPTION
**Pull request by kpbochenek**
fixes #494

### CODE REVIEW

File: tests/shared/src/test/scala/scala/meta/tests/parsers/dotty/MinorDottySuite.scala [hunk 1]
```diff
@@ -372,4 +372,46 @@ class MinorDottySuite extends BaseDottySuite {
       )
     )
   }
+
+  test("error on repeated byname class parameter") {
```
Comment (tgodzik) [inline tests/shared/src/test/scala/scala/meta/tests/parsers/dotty/MinorDottySuite.scala:376]: This is not an error though?

File: tests/shared/src/test/scala/scala/meta/tests/parsers/VarargParameterSuite.scala [hunk 2]
```diff
@@ -126,7 +130,7 @@ class VarargParameterSuite extends ParseSuite {
     val error = intercept[parsers.ParseException] {
       templStat(definition)
     }
-    assert(error.getMessage.contains(expected))
+    assert(clue(error).getMessage.contains(clue(expected)))
```
Comment (tgodzik) [inline tests/shared/src/test/scala/scala/meta/tests/parsers/VarargParameterSuite.scala:133]: We shouldn't need to change this if it was working before.

**Review by tgodzik — CHANGES_REQUESTED** [review]

File: tests/shared/src/test/scala/scala/meta/tests/parsers/dotty/MinorDottySuite.scala (hunk 1 above)
Comment (kpbochenek) [reply tests/shared/src/test/scala/scala/meta/tests/parsers/dotty/MinorDottySuite.scala]: right, changed :+1:

File: tests/shared/src/test/scala/scala/meta/tests/parsers/VarargParameterSuite.scala (hunk 2 above)
Comment (kpbochenek) [reply tests/shared/src/test/scala/scala/meta/tests/parsers/VarargParameterSuite.scala]: reverted

**Review by tgodzik — APPROVED** [review]
LGTM!

**Pull request merged**